app/models/tag.rb

```diff
@@ -688,10 +689,12 @@ def recategorize(new_type)
   def reindex_taggables
     work_ids = all_filtered_work_ids
     series_ids = all_filtered_series_ids
+    external_work_ids = all_filtered_external_work_ids
     bookmark_ids = all_bookmark_ids
     yield if block_given?
     reindex_all_works(work_ids)
     reindex_all_series(series_ids)
+    reindex_all_external_works(all_filtered_external_work_ids)
```

Comment:
This should use the `external_work_ids` instead of calling the `all_filtered_external_work_ids` method a second time.